### ISSUE TITLE
SP2 DD lacks many later messages

### ISSUE THREAD

**Issue by gbirchmeier** [connamara/quickfixn#635]
The newest message in SP2 is "CG", added in Extension Pack 105.

It currently goes up to EB.

[Had a mail list user confused why he couldn't crack DQ.](http://lists.quickfixn.com/pipermail/quickfixn-quickfixn.com/2020q3/001164.html)

**Comment by ericvergnaud** [comment]
Hi, is there a plan to address this ? (it's been around for almost 6 years?)

**Comment by gbirchmeier** [comment]
The short answer is: There hasn't been a vocal demand for it, so it hasn't been a priority.  (Certainly no one's offered to contribute anything either.)

Also, it's reasonably easy to edit the DD and then regenerate the message-package source and create a custom dll that will then have classes for whatever messages and fields you need.  (Not as easy as just grabbing a nuget, of course, but it's not difficult at all.)  Given that, in practice, *all* counterparties make modification to their DDs, I think some users might be doing that.  I would be happy to assist you, as it would give me a reason to write a good HOWTO doc about that.

**With all that said**, this is an improvement I want to make.  I'm vaguely aware that the FIX Org has a machine-readable message/field DB now, and I can probably use that  as the basis for a new release, but I haven't had time to fully dig into it yet.

**Comment by ericvergnaud** [comment]
I can provide a PR if you provide guidance…So I guess this is maybe the opportunity to write a HOWTO ?

**Comment by ericvergnaud** [comment]
Looks like the code could be generated using Orchestra?
For example [this xml file](https://raw.githubusercontent.com/FIXTradingCommunity/fix-orchestra/refs/heads/master/repository/src/test/resources/examples/mit_2016.xml) seems a good example starting point.
But that's a lot of work in perspective.

It seems they also have a tool to generate QuickFix DDs from Orchestra specs? 
See https://claude.ai/share/ea8bd35c-6b41-4eb1-ba59-b791aec778aa
Is the QuickFix/n DD the same as the QuickFix/j DD ?

**Comment by gbirchmeier** [comment]
Yes, Orchestra was what I was alluding to earlier.  And yes, the QF/j DDs are pretty much the same as hte QF/n DDs (there may be some divergences in content over the years, but the structure is the same).

Orchestra is something I want to learn more about, but I haven't gotten to it yet.

What drives me crazy about Orchestra is that, so far, I cannot find the machine-readable specification for the FIX messages.  The pages talk all about how to define a DD, but I'll be damned if I can actually find the machine-readable source that tells me e.g. NewOrderSingle is "D".

I did find this, which is nice: https://orchestrahub.org/-/fix-latest -- but where did he get this data from?  Why isn't it easy to find from the Orchestra docs?

(Or am I missing something right in front of my face?  That's entirely possible.)

PS  You posted a broken Claude link.  Also, please don't send Claude chats; if Claude can say something, then it can also give a source for it.  Post the source.

**Comment by gbirchmeier** [comment]
Ah, I think I finally found it:
https://github.com/FIXTradingCommunity/orchestrations/tree/master/FIX%20Standard

**Comment by ericvergnaud** [comment]
I've fixed the claude link, I had forgotten to request a shared link. (not sure what you mean by "post the source") ?

**Comment by ericvergnaud** [comment]
I guess generating from Orchestra would be ultimately better since we could then generate the conditional validation code on top of the message structures.
But on the short term, simply updating the DD by regenerating it using Orchestra might be a big -cheap- step forward ? 
I can give it a try if you think it's a good approach.
I would still need a HOWTO in order to regenerate the code from the DD.
